cclib/method/ddec.py

```diff
@@ -590,9 +626,73 @@ def condition_densities(self):
             # Make tau monotonic decreasing
             self.tau[atomi] = numpy.maximum.accumulate(self.tau[atomi][::-1])[::-1]
 
-    def _phiai(self, ya, bigphiAI, atomi):
-        """ Evaluate phi_A^I based on equation S100
+    def calculate_w_and_u(self):
+        """ Calculate weights placed on each integration grid point
+            [STEP 4-7]
         """
+        # From equation 67, w_A(r_A) = self._cond_density
+        # From equation 8, W(r) = self.rho_cond
+        xshape, yshape, zshape = self.chgdensity.data.shape
+        atoms = len(self.data.atomnos)
```

Comment:
`atoms` doesn't indicate that this is a count, prefer
```suggestion
        natoms = len(self.data.atomnos)
```

Reply:
Yes. I've made the necessary changes. Thanks!